pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -254,6 +254,34 @@ func TestPipelineSpec_Validate_Failure(t *testing.T) {
 			Message: `expected exactly one, got both`,
 			Paths:   []string{"tasks[0].conditions", "tasks[0].when"},
 		},
+	}, {
+		name: "invalid pipeline with one pipeline task having both conditions and when expressions in finally",
+		ps: &PipelineSpec{
+			Description: "this is an invalid pipeline with invalid pipeline task",
+			Tasks: []PipelineTask{{
+				Name:    "invalid-pipeline-task",
+				TaskRef: &TaskRef{Name: "foo-task"},
+			}},
+			Finally: []PipelineTask{{
+				Name:    "invalid-pipeline-final-task",
+				TaskRef: &TaskRef{Name: "foo-task"},
+				WhenExpressions: []WhenExpression{{
+					Input:    "foo",
+					Operator: selection.In,
+					Values:   []string{"bar"},
+				}},
+				Conditions: []PipelineTaskCondition{{
+					ConditionRef: "some-condition",
+				}},
+			}},
+		},
+		expectedError: *apis.ErrGeneric("").Also(&apis.FieldError{
```

Comment:
If conditions are not allowed in finally tasks at all, I think it's almost more confusing to have the error message state that you can't have both conditions and when expressions.

Reply:
I see what you're saying, but a user would get both error messages, that `conditions` is not allowed in `finally` and that both `conditions` and `when` expressions cannot be together

Reply:
it's all good, wouldn't want to make it more confusing to users when trying to clean up the code to remove duplications